compiler/magicsys.nim

```diff
@@ -79,7 +79,9 @@ proc getSysType*(g: ModuleGraph; info: TLineInfo; kind: TTypeKind): PType =
     else: internalError(g.config, "request for typekind: " & $kind)
     g.sysTypes[kind] = result
   if result.kind != kind:
-    internalError(g.config, "wanted: " & $kind & " got: " & $result.kind)
+    if kind == tyFloat64 and result.kind == tyFloat: discard # because of aliasing
```

Comment:
There is no such aliasing, ``tyFloat64`` is unused.

Reply:
huh?
`nim c main.nim`
with:
```nim
static: echo "ok1"
static: echo float64
static: echo "ok2"
static: echo float64
static: echo "ok3"
static: echo 1.0
```
outputs:
`in getSysType:tyFloat64` (82 times!)
followed by:
```
ok1
float64
ok2
float64
ok3
in getSysType:tyFloat64
1.0
```

Reply:
Ok, but why:

```nim

type
  float* {.magic: Float.}     ## Default floating point type.
  float32* {.magic: Float32.} ## 32 bit floating point type.
  float64* {.magic: Float.}   ## 64 bit floating point type.
```

Reply:
because of this:
```nim
  of nkFloat64Lit, nkFloatLit:
    if result.typ == nil: result.typ = getSysType(c.graph, n.info, tyFloat64)
```

Reply:
that's to be fixed then.

Reply:
can be fixed in subsequent PR, this is still a strict improvement over status quo...